compare-evidence-strings/compare.sh

```diff
@@ -1,130 +1,210 @@
+# A function to extract all unique identifying fields from the evidence strings. The fields being extracted are:
 # * ClinVar RCV accession
+# * Phenotype
+# * Allele origin
 # * Variant ID (rsID or, if absent, RCV accession)
 # * Ensembl gene ID
-# * Functional consequence SO code
 extract_fields () {
   ./jq '
     .unique_association_fields.clinvarAccession + "|" +
+    .unique_association_fields.phenotype + "|" +
+    .unique_association_fields.alleleOrigin + "|" +
     .unique_association_fields.variant_id + "|" +
-    .unique_association_fields.gene + "|" +
-    .evidence.gene2variant.functional_consequence
-  ' < "$1" \
-  | tr -d '"' | tr '|' '\t' \
+    .unique_association_fields.gene
+  ' < "$1" | tr -d '"' > "$2"
+}
+
+# Computes a word diff between two files using git diff
+compute_git_diff () {
+  # The --no-index option is important, because otherwise git will refuse to compare the files if you're running this
+  # script from right inside the repository (because the files are untracked).
+  git diff \
+  --minimal \
+  -U0 \
+  --color=always \
+  --word-diff=color \
+  --no-index \
+  "$1" "$2"
+}
+
+# Extract only the functional consequence from the evidence string
+extract_functional_consequences () {
+  ./jq '.evidence.gene2variant.functional_consequence' \
+  | tr -d '"' \
   | sed -e 's|http://purl.obolibrary.org/obo/||g' \
-        -e 's|http://identifiers.org/clinvar.record/||g' \
-        -e 's|http://identifiers.org/dbsnp/||g' \
-        -e 's|http://www.ncbi.nlm.nih.gov/clinvar/||g' \
-  > "$2"
+        -e 's|http://targetvalidation.org/sequence/||g'
 }
 
-echo "Set up environment & parse parameters"
-export -f sort_keys extract_fields
-# To ensure that the sort results are consistent, set the sort order locale directly
+export -f sort_keys extract_fields compute_git_diff extract_functional_consequences
+
+
+
+########################################################################################################################
+echo "Preparation"
+
+echo "  Set up environment and parse parameters"
+# To ensure that the sort results are consistent, set the sort order locale explicitly
 export LC_COLLATE=C
 # The realpath is required to make the paths work after the working directory change
 OLD_EVIDENCE_STRINGS=$(realpath "$1")
 NEW_EVIDENCE_STRINGS=$(realpath "$2")
 mkdir comparison && cd comparison || exit 1
 
-echo "Install JQ — a command line JSON processor"
+echo "  Install JQ — a command line JSON processor"
 wget -q -O jq https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64
 chmod a+x jq
 
-echo "Install aha — HTML report generator"
+echo "  Install aha — HTML report generator"
 wget -q https://github.com/theZiz/aha/archive/0.5.zip
-unzip -q 0.5.zip && cd aha-0.5 && make && mv aha ../ && cd .. && rm -rf aha-0.5 0.5.zip
+unzip -q 0.5.zip && cd aha-0.5 && make &>/dev/null && mv aha ../ && cd .. && rm -rf aha-0.5 0.5.zip
```

Comment:
Would suggest the environment preparation section to a separate script and invoke it...

Reply:
I agree, there's something that feels wrong about installing jq and aha everytime you call ./compare.sh

Reply:
Agreed, fixed in https://github.com/EBIvariation/eva-cttv-pipeline/pull/100/commits/82bbfda794e1405bb97e4cf79516bd9d4b888370